client/src/main/kotlin/no/nav/su/se/bakover/client/person/AdGraphApiClient.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.su.se.bakover.client.person
+
+import arrow.core.Either
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.github.kittinunf.fuel.core.extensions.authentication
+import com.github.kittinunf.fuel.httpGet
+import no.nav.su.se.bakover.client.azure.OAuth
+import no.nav.su.se.bakover.common.objectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+data class AdGraphResponse(
+    val onPremisesSamAccountName: String,
+    val displayName: String,
+    val givenName: String,
+    val mail: String,
+    val officeLocation: String,
+    val surname: String,
+    val userPrincipalName: String,
+    val id: String,
+    val jobTitle: String
+)
+
+class AdGraphApiClient(
+    private val exchange: OAuth
+) {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+
+    fun hent(userToken: String): Either<Pair<Int, String>, AdGraphResponse> {
+        val onBehalfOfToken = exchange.onBehalfOFToken(userToken, "https://graph.microsoft.com")
```

Comment:
Er det behov for onBehalfOfToken her? Regner med du har testet ut, bare nysgjerrig.

Reply:
Basert på det andre jeg har sett på så er det det man skal bruke ja. Kan sjekke om det går an uten.